src/cryptonote_core/miner.cpp

```diff
@@ -58,6 +58,10 @@ namespace cryptonote
     const command_line::arg_descriptor<std::string> arg_extra_messages =  {"extra-messages-file", "Specify file for extra messages to include into coinbase transactions", "", true};
     const command_line::arg_descriptor<std::string> arg_start_mining =    {"start-mining", "Specify wallet address to mining for", "", true};
     const command_line::arg_descriptor<uint32_t>      arg_mining_threads =  {"mining-threads", "Specify mining threads count", 0, true};
+    const command_line::arg_descriptor<bool>        arg_bg_mining_enable =  {"bg-mining-enable", "Specify mining threads count", 1, true};
+    const command_line::arg_descriptor<uint64_t>    arg_bg_mining_idle_interval_seconds =  {"bg-mining-idle-interval", "Specify lookback interval in seconds for determining idle state", 2, true};
+    const command_line::arg_descriptor<uint8_t>     arg_bg_mining_idle_threshold_percentage =  {"bg-mining-idle-threshold", "Specify minimum avg idle percentage over lookback interval", 3, true};
```

Comment:
I wanted to call this out separately ... this means by default background mining initiates when the users system is 3% idle ... ? That seems fairly low to me, but any default value is arbitrary I suppose.